draft-irtf-cfrg-hybrid-kems.md

```diff
@@ -679,6 +715,37 @@ def Decaps(dk, ct):
     return ss_H
 ~~~
 
+## Separate Key Generation
+
+The hybrid KEM framework described above prescribes a "shared seed" process for
+keypair generation, in which a shared seed is expanded into seeds for the
+individual constituent algorithms. This shared seed is used as the decapsulation
+key for the hybrid KEM and expanded into the two per-constituent decapsulation
+keys as necessary.
+
+In some deployment environments, it is not possible to instantiate this process.
+Some implementations of constituent algorithms do not support the
+`DeriveKeyPair` function, only `GenerateKeyPair`.  Likewise in the nominal group
+case, a (scalar, group element) pair will only be generated when the scalar is
+generated internal to the implementation.
+
+An implementation of a hybrid KEM in such environemnts MAY deviate from the
+above description in the following ways:
+
+* `DeriveKeyPair` is not implemented.
+* The decapsulation key returned by `GenerateKeyPair` and consumed by `Decaps`
+  is a tuple `(dk_PQ, dk_T)` of per-constituent decapsulation keys (or
+  pointers/handles to keys).
+
+These deviations have both interoperability and security impacts.
```

Comment:
Probably needs a paragraph about FIPS compliance and how FIPS 203 requires the seed to be direct output of a DRBG, so it will be impossible to do a shared seed keygen in a crypto library that is required to run in FIPS mode.